Server/Application/Application.m

```diff
@@ -103,7 +103,17 @@ + (void)launchAppWithBundleId:(NSString *_Nullable)bundleId
     }
 
     application.launchArguments = launchArgs ?: @[];
-    application.launchEnvironment = environment ?: @{};
+    if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 10.3) {
+        if (environment != nil && [environment objectForKey:@"DYLD_INSERT_LIBRARIES"] != nil) {
+            NSMutableDictionary *environmentCopy = [environment mutableCopy];
+            NSString *dynLibs = [[environment objectForKey:@"DYLD_INSERT_LIBRARIES"]
+                                              stringByAppendingString:@":/Developer/usr/lib/libXCTTargetBootstrapInject.dylib"];
+            [environmentCopy setObject:dynLibs forKey:@"DYLD_INSERT_LIBRARIES"];
+            application.launchEnvironment = environmentCopy;
+        }
+    } else {
+        application.launchEnvironment = environment ?: @{};
+    }
```

Comment:
what if the environment is nil or does not have the DYLD_INSERT_LIBRARIES key?

in that case we should end up and environment where the key DYLD_INSERT_LIBRARIES has the value of /Developer/usr/lib/libXCTTargetBootstrapInject.dylib